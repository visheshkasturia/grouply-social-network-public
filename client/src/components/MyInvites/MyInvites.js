import React, { useState, useEffect } from 'react';
import MyInvitesCard from './MyInvitesCard.js/MyInvitesCard';
/* eslint-disable */
function MyInvites({ user }) {
  const [invitationList, setInvitationList] = useState(user.invitations)

  const getChangedUser = (data) => {
    setInvitationList(data);
  } 
 
  useEffect(() => {
  }, [invitationList])

  return (
    <div className="my-invites-container">
        { invitationList.map((invite) => (
            < MyInvitesCard key={invite} user={user} invite = {invite} getChangedUser={getChangedUser} />
        ))
      }
    </div>
  )
}

export default MyInvites;
