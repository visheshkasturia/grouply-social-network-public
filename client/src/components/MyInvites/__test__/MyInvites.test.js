import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MyInvites from '../MyInvites'

const admin = 'abc'
const member = 'test1'
const post = '1'
const tag = 'tag'
const group = {
  groupName: "test",
  description: "desc",
  admins: [admin],
  allMembers: [member],
  allPosts: [post],
  tags: [tag]
}

const user = {
  _id: '1',
  invitations: ['abc'],
}

it('MyInvites renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MyInvites user={user} />, div);
});

it('Members matches snapshot', () => {
  const tree = renderer.create(<MyInvites user={user} />).toJSON();
  expect(tree).toMatchSnapshot();
});
