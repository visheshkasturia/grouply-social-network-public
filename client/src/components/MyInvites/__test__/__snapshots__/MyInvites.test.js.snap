// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members matches snapshot 1`] = `
<div
  className="my-invites-container"
>
  <div
    className="invite-card-container"
  >
    <div
      className="invite-card-groupName"
    >
      abc
    </div>
    <div
      className="invite-card-accept"
    >
      <button
        onClick={[Function]}
        type="button"
      >
         Accept 
      </button>
    </div>
    <div
      className="invite-card-reject"
    >
      <button
        onClick={[Function]}
        type="button"
      >
         Reject 
      </button>
    </div>
  </div>
</div>
`;
